File: src/main/java/com/myapp/crud/appgen/home/HomeService.java
package com.myapp.crud.appgen.home;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import com.myapp.crud.appgen.codeGroups.CodeGroups;

@Service
public class HomeService {
	private HomeDAO homeDAO;

	public HomeService() {
	}

	@Autowired
	public HomeService(HomeDAO homeDAO) {
		this.homeDAO = homeDAO;
	}

	public Page findByAll(Pageable pageable) {
		return homeDAO.findAll(pageable);
//		return homeDAO.findAll(PageRequest.of(page, size));
	}
	
	public Home findById(Long id) {
		if(id != null && id.longValue() == -1) {
			return new Home();
		}
		return homeDAO.findById(id).get();
	}

	public Home save(Home home) {
		return homeDAO.save(home);
	}

	public void deleteById(Long id) {
		homeDAO.deleteById(id);
	}
	public List<CodeGroups> findByDistinctCode(){
		return homeDAO.findByDistinctCode();
	}
	public List<Home> findByUrlStr(String urlStr) {
		return homeDAO.findByUrlStr(urlStr);
	}
	public List<Home> findByName(String name) {
		return homeDAO.findByName(name);
	}

}

File: ui/react/src/api/home/HomeService.js
import axios from "axios";
import { URL_BASE } from '../Utils.js'

class HomeService {
 getList = () => {
  const theurl = `${URL_BASE}/home/list`
  return axios.get(theurl);
 }
 get = (id) => {
  const theurl = `${URL_BASE}/home/${id}`
  console.log(`[HomeService.get] theurl ${theurl}`);
  return axios.get(theurl);
 }
 save = (data) => {
  const theurl = `${URL_BASE}/home/save`
  console.log(`[HomeService.save] theurl ${theurl}`);
  console.log(`[HomeService.save] data`, data);
  return axios.post(theurl, data);
 }
}

export default new HomeService();